code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -948,14 +948,13 @@
 		for(var/i in connected_ai.aicamera.stored)
 			aicamera.stored[i] = TRUE
 
-/mob/living/silicon/robot/proc/charge(datum/source, amount, repairs, sendmats)
+/mob/living/silicon/robot/proc/charge(datum/source, datum/callback/charge_cell, seconds_per_tick, repairs, sendmats)
 	SIGNAL_HANDLER
+	charge_cell.Invoke(cell, seconds_per_tick)
 	if(model)
-		model.respawn_consumable(src, amount * 0.005)
+		model.respawn_consumable(src, cell.use(STANDARD_CELL_CHARGE * 0.001))
```

Comment:
I hate this proc. `respawn_consumable()` does not return an amount which would tell us how much more energy it needs. We just keep feeding it till the end of days i guess.

Hopefully someone comes up with a good solution for this